File: frontend/src/interfaces/OrderPromotion.ts
import type { Order } from "./Order";
import type { Promotion } from "./Promotion";

export interface OrderPromotion {
  ID: number;
  discount_amount: number;

  order_id: number;
  order?: Order;

  promotion_id: number;
  promotion?: Promotion;
}

export interface CreateOrderPromotionRequest {
  discount_amount: number;
  order_id: number;
  promotion_id: number;
}

File: backend/entity/order_promotion.go
// entity/order_promotion.go
package entity

import "gorm.io/gorm"

// ตารางกลาง Order <> Promotion พร้อม field ส่วนลดที่ใช้จริงกับออเดอร์นั้น
type OrderPromotion struct {
	gorm.Model
	DiscountAmount float64 `json:"discount_amount"`

	OrderID uint   `json:"order_id"`
	Order   *Order `gorm:"foreignKey:OrderID" json:"order,omitempty"`

	PromotionID uint       `json:"promotion_id"`
	Promotion   *Promotion `gorm:"foreignKey:PromotionID" json:"promotion,omitempty"`
}
